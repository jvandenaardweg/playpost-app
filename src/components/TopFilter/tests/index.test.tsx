import React from 'react';
import { render, RenderAPI } from 'react-native-testing-library';

import { TopFilter } from '../index';

const onSelectHandler = jest.fn()
const defaultProps = {
  filters: [
    {
      label: 'Country',
      options: ['All', 'EN', 'NL', 'DE'],
      selectedOption: 'All',
      onSelect: onSelectHandler
    }
  ]
}

describe('TopFilter', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;

    beforeEach(() => {
      const props = {
        ...defaultProps
      }
      wrapper = render(<TopFilter {...props} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should render the correct filter label', () => {
      expect(wrapper.getByTestId('TopFilter-Text-label').props.children).toBe(defaultProps.filters[0].label);
    });

    it('should render the correct amount of filter options', () => {
      expect(wrapper.queryAllByTestId('TopFilter-View-ButtonTiny-container')).toHaveLength(defaultProps.filters[0].options.length);
    });
  });
});
