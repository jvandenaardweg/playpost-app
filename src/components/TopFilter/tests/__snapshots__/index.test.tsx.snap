// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopFilter rendering should render correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderBottomColor": "#e5e5e5",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "paddingBottom": 12,
        "paddingTop": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#040404",
            "fontFamily": "Inter-SemiBold",
            "fontSize": 13,
            "lineHeight": 16,
          },
          Object {
            "marginLeft": 12,
            "marginRight": 12,
            "width": 56,
          },
        ]
      }
      testID="TopFilter-Text-label"
    >
      Country
    </Text>
    <RCTScrollView
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
      decelerationRate={0}
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      testID="TopFilter-ScrollView"
    >
      <View>
        <View
          style={
            Object {
              "marginRight": 6,
            }
          }
          testID="TopFilter-View-ButtonTiny-container"
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 28,
                  "height": 28,
                },
                Object {
                  "backgroundColor": "#D9E9FD",
                },
              ]
            }
            testID="ButtonTiny-Button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 28,
                    "flexDirection": "row",
                    "height": 28,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                  },
                  Object {
                    "backgroundColor": "#D9E9FD",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 13,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "#0066FF",
                    },
                  ]
                }
              >
                All
              </Text>
              <View
                style={
                  Object {
                    "marginLeft": 6,
                    "marginRight": -1,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#0066FF",
                        "fontSize": 10,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome5Free-Solid",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {
                        "fontWeight": "900",
                      },
                    ]
                  }
                >
                  ï€Œ
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginRight": 6,
            }
          }
          testID="TopFilter-View-ButtonTiny-container"
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 28,
                  "height": 28,
                },
                Object {
                  "backgroundColor": "#EEEEEE",
                },
              ]
            }
            testID="ButtonTiny-Button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 28,
                    "flexDirection": "row",
                    "height": 28,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                  },
                  Object {
                    "backgroundColor": "#EEEEEE",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 13,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "#040404",
                    },
                  ]
                }
              >
                EN
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginRight": 6,
            }
          }
          testID="TopFilter-View-ButtonTiny-container"
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 28,
                  "height": 28,
                },
                Object {
                  "backgroundColor": "#EEEEEE",
                },
              ]
            }
            testID="ButtonTiny-Button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 28,
                    "flexDirection": "row",
                    "height": 28,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                  },
                  Object {
                    "backgroundColor": "#EEEEEE",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 13,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "#040404",
                    },
                  ]
                }
              >
                NL
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "marginRight": 6,
            }
          }
          testID="TopFilter-View-ButtonTiny-container"
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "borderRadius": 28,
                  "height": 28,
                },
                Object {
                  "backgroundColor": "#EEEEEE",
                },
              ]
            }
            testID="ButtonTiny-Button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "borderRadius": 28,
                    "flexDirection": "row",
                    "height": 28,
                    "paddingLeft": 12,
                    "paddingRight": 12,
                  },
                  Object {
                    "backgroundColor": "#EEEEEE",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#040404",
                      "fontFamily": "Inter-SemiBold",
                      "fontSize": 13,
                      "lineHeight": 16,
                    },
                    Object {
                      "color": "#040404",
                    },
                  ]
                }
              >
                DE
              </Text>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
